File: exercises/practice/matrix/matrix.zig
const std = @import("std");
const mem = std.mem;

/// Returns the selected row of the matrix.
pub fn row(allocator: mem.Allocator, s: []const u8, index: i32) ![]i16 {
    _ = allocator;
    _ = s;
    _ = index;
    @compileError("please implement the row function");
}

/// Returns the selected column of the matrix.
pub fn column(allocator: mem.Allocator, s: []const u8, index: i32) ![]i16 {
    _ = allocator;
    _ = s;
    _ = index;
    @compileError("please implement the column function");
}

File: generators/exercises/matrix.py
from lib import zstr

HEADER = "const row = matrix.row;\nconst column = matrix.column;\n"


def gen_case(case):
    prop = case["property"]  # "row" or "column"
    s = case["input"]["string"]
    index = case["input"]["index"]
    expected = ", ".join(str(v) for v in case["expected"])
    return (
        f"    const expected = &[_]i16{{ {expected} }};\n"
        f"    const s = {zstr(s)};\n"
        f"    const actual = try {prop}(testing.allocator, s, {index});\n"
        f"    defer testing.allocator.free(actual);\n"
        f"    try testing.expectEqualSlices(i16, expected, actual);\n"
    )
